internal/layer2/arp.go

```diff
@@ -97,7 +97,10 @@ func (a *arpResponder) processRequest() dropReason {
 	}
 
 	// Ignore ARP requests that the announcer tells us to ignore.
-	if reason := a.announce(pkt.TargetIP); reason != dropReasonNone {
+	if reason := a.announce(pkt.TargetIP, a.intf); reason != dropReasonNone {
```

Comment:
let's avoid nested ifs:

```suggestion
	reason := a.announce(pkt.TargetIP, a.intf)
	if reason == dropReasonNotMatchInterface {
	log
	}
	if reason != dropReasonNone {
	   return reason
	}
```